HPXMLtoOpenStudio/resources/airflow.rb

```diff
@@ -1479,155 +1724,128 @@ def self.apply_infiltration_and_ventilation_fans(model, weather, vent_fans_mech,
-      infil_program.addLine("Set #{erv_lat_load_actuator.name} = 0.0")
-
-      # ERV/HRV EMS load model
-      # E+ ERV model is using standard density for MFR calculation, caused discrepancy with other system types.
-      # Therefore ERV is modeled within EMS infiltration program
-
-      # Air property at inlet nodes in two sides of ERV
-      # Common part across multiple erv/hrv programs, create only once
-      infil_program.addLine("Set ERVSupInPb = #{@pbar_sensor.name}") # oa barometric pressure
-      infil_program.addLine("Set ERVSupInTemp = #{@tout_sensor.name}") # oa db temperature
-      infil_program.addLine("Set ERVSupInW = #{@wout_sensor.name}")   # oa humidity ratio
-      infil_program.addLine('Set ERVSupRho = (@RhoAirFnPbTdbW ERVSupInPb ERVSupInTemp ERVSupInW)')
-      infil_program.addLine('Set ERVSupCp = (@CpAirFnW ERVSupInW)')
-      infil_program.addLine('Set ERVSupInEnth = (@HFnTdbW ERVSupInTemp ERVSupInW)')
-
-      infil_program.addLine("Set ERVSecInTemp = #{@tin_sensor.name}") # zone air temperature
-      infil_program.addLine("Set ERVSecInW = #{@win_sensor.name}") # zone air humidity ratio
-      infil_program.addLine('Set ERVSecCp = (@CpAirFnW ERVSecInW)')
-      infil_program.addLine('Set ERVSecInEnth = (@HFnTdbW ERVSecInTemp ERVSecInW)')
-    end
-
-    # Calculate cfm weighted average effectiveness for hvac sizing
-    # Please review this, and its integration with hvac_sizing
-    weighted_vent_mech_lat_eff = 0.0
-    weighted_vent_mech_apparent_sens_eff = 0.0
-
-    # Apply ERV/HRV mechanical ventilation
-    vent_mech_erv_hrv.each do |vent_mech|
-      vent_mech_sens_eff, vent_mech_lat_eff, vent_mech_apparent_sens_eff = calc_hrv_erv_effectiveness(vent_mech)
-      # No need to add balanced system here to average because their effectivenesses are all 0.0
-      weighted_vent_mech_lat_eff += vent_mech.average_flow_rate / tot_bal_cfm * vent_mech_lat_eff
-      weighted_vent_mech_apparent_sens_eff += vent_mech.average_flow_rate / tot_bal_cfm * vent_mech_apparent_sens_eff
-
-      infil_program = apply_erv_hrv_load_to_infil_program(model, infil_program, vent_mech.average_flow_rate, vent_mech_sens_eff, vent_mech_lat_eff, erv_sens_load_actuator, erv_lat_load_actuator)
-    end
-
-    model.getBuilding.additionalProperties.setFeature(Constants.SizingInfoMechVentLatentEffectiveness, weighted_vent_mech_lat_eff)
-    model.getBuilding.additionalProperties.setFeature(Constants.SizingInfoMechVentApparentSensibleEffectiveness, weighted_vent_mech_apparent_sens_eff)
+    # Common variable and load actuators across multiple mech vent calculations, create only once
+    infil_program, fan_sens_load_actuator, fan_lat_load_actuator = setup_mech_vent_vars_actuators(model: model, program: infil_program)
 
     # Apply CFIS
-    infil_program.addLine('Set QWHV_cfis = 0.0')
     infil_program.addLine("Set #{cfis_fan_actuator.name} = 0.0")
+    infil_program = apply_cfis(infil_program, vent_mech_cfis_tot, cfis_fan_actuator)
```

Comment:
No need to `infil_program = `. Comment applies to additional lines of code below.